keps/sig-apps/0003-kubernetes-application-api.md

```diff
@@ -0,0 +1,1108 @@
+kubectl to make use the of the Application Kind.
+
+### User Stories
+
+#### Application Creation
+As an application maintainer or tool developer, I can create an application that is interoperable with other tools and 
+has first class support in Kuberentes UIs.
+
+#### Application View
+As an application user or UI designer, I have a standard way to view the components and status of Kubernetes 
+applications.
+
+#### Application Health
+As an application developer, I can surface a basic health check for UIs to consume, and, as a UI developer, I have a 
+standard way to consume application level health checks.
+
+#### Application Update
+As an application developer, maintainer, or administrator, I can update the configuration of my application.
+
+#### Application List
+As an application developer, maintainer or administrator, I can list the applications that are currently installed.
+
+#### Application Watch
+As an application developer, maintainer or administrator, I can watch the applications in a cluster for changes.
+
+#### Application Deletion
+As a user, I can delete my application, and when I delete my Application, its corresponding components are deleted.
+
+### Implementation
+This section describes the implementation of the feature, including the API, the associated controller, and the CLI 
+and UI interaction.
+
+### API
+The go code below describes the Application CRD. 
+
+```go
+
+// ApplicationStatus contains the status of an Application. 
+type ApplicationStatus struct {
+    // ObservedGeneration is used by the Application Controller to report the last Generation of an Application 
+    // that it has observed.
+    ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"int64,1,opt,name=observedGeneration"`
+    
+    // Installed is used by the Application Controller to report the installed Components and Dependencies of  
+    // an Application.
+    Installed []string `json:"installed,omitempty" protobuf:"bytes,2,opt,name=installed"`
+    
+    // Ready is used to report the readiness of an Application. If an Application contains an ApplicationHealthCheck, 
+    // this field corresponds directly to the Readiness of the Pod generated from that health check. If the 
+    // Application does not contain an ApplicationHealthCheck, this field is set to Unknown.
+    Ready ConditionStatus json:`"ready,omitempty" protobuf:"bytes,3,opt,name=ready"`
+}
+
+// ApplicationSpec is the specification object for an Application. It specifies the Application's components, 
+// dependencies, and an optional health check. Application components can be retrieved using the Applications 
+// spec.selector to select the resources that match from the components GroupKind at the client's desired version.
+type ApplicationSpec struct {
+    // Type is the type of the application (e.g. WordPress, MySQL, Cassandra).
+    Type string `json:"type" protobuf:"bytes,1,name=type"`
```

Comment:
How is the type different from the name (in the metadata)? In your example below, they both have the same value, what's the purpose? When do you expect them to be different, etc?

Reply:
Based on the examples, it seems like the field is intended to be descriptive (not programatically consumed) and captures something about the source of the application instance (open source project, product name, etc.). However, "Type" seems to hint at having a programmatic meaning. If the intent is descriptive, consider alternatives:
- Title
- Brand

Reply:
Is there an explicit usage of this field? As there is no clear definition of "type" and user could use it arbitrarily which makes it more like a metadata.